File: packages/contracts/python/noa_contracts/inspiration_item.py
from __future__ import annotations

from typing import Literal, Any
from pydantic import BaseModel


class InspirationItem(BaseModel):
    schema_version: Literal["1.0"] = "1.0"
    id: str
    board_id: str
    source_type: Literal["upload", "curated", "url_import"]
    storage_key: str | None
    source_url: str | None
    thumbnail_key: str | None
    embedding: list[float] | None
    embedding_model: Literal["clip-vit-l-14"] | None
    weight: float
    sentiment: Literal["positive", "negative"]
    annotations: list[dict[str, Any]]
    highlighted_elements: list[dict[str, Any]]
    rejected_elements: list[dict[str, Any]]
    upload_status: Literal["pending", "ready", "failed"]
    created_at: str

File: packages/contracts/v1/inspiration-item.ts
import type { UUID, ISO8601 } from "./common.js";

export interface ItemAnnotation {
  [key: string]: unknown;
}

export interface HighlightedElement {
  [key: string]: unknown;
}

export interface RejectedElement {
  [key: string]: unknown;
}

export interface InspirationItem {
  schema_version: "1.0";
  id: UUID;
  board_id: UUID;
  source_type: "upload" | "curated" | "url_import";
  storage_key: string | null;
  source_url: string | null;
  thumbnail_key: string | null;
  embedding: number[] | null;
  embedding_model: "clip-vit-l-14" | null;
  weight: number;
  sentiment: "positive" | "negative";
  annotations: ItemAnnotation[];
  highlighted_elements: HighlightedElement[];
  rejected_elements: RejectedElement[];
  upload_status: "pending" | "ready" | "failed";
  created_at: ISO8601;
}
